### PR TITLE
Add `precision`, `amp_backend`, and `amp_level` as properties of `PrecisionPlugin`

### PR DESCRIPTION
**Pull request by daniellepintz**
## What does this PR do?
Part of https://github.com/PyTorchLightning/pytorch-lightning/issues/11814

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/plugins/precision/mixed.py [hunk 1]
```diff
@@ -23,4 +23,7 @@ class MixedPrecisionPlugin(PrecisionPlugin):
     """Base Class for mixed precision."""
 
     backend: "AMPType"
```
Comment (four4fish) [inline pytorch_lightning/plugins/precision/mixed.py:25]: I think we can get rid of this now

File: pytorch_lightning/plugins/precision/apex_amp.py [hunk 2]
```diff
@@ -98,3 +98,11 @@ def state_dict(self) -> Dict[str, Any]:
 
     def load_state_dict(self, state_dict: Dict[str, Any]) -> None:
         amp.load_state_dict(state_dict)
+
+    @property
+    def amp_backend(self) -> Optional[AMPType]:
```
Comment (four4fish) [inline pytorch_lightning/plugins/precision/apex_amp.py:103]: I think we can remove L33?

File: pytorch_lightning/plugins/precision/precision_plugin.py [hunk 3]
```diff
@@ -275,3 +273,15 @@ def on_load_checkpoint(self, checkpoint: Dict[str, Any]) -> None:
...
+    def precision(self) -> Union[str, int]:
+        return 32
+
+    @property
+    def amp_backend(self) -> Optional[AMPType]:
+        return None
+
+    @property
+    def amp_level(self) -> Optional[str]:
+        return None
```
Comment (four4fish) [inline pytorch_lightning/plugins/precision/precision_plugin.py:287]: I'm not sure it make sense to include this in the base class. 
The TPU, IPU, DoublePrecision (which inheritance from the base class PrecisionPlugin) doesn't have amp. I think we can add in MixPrecisionPlugin. Maybe deepSpeedPrecisionPlugin can inherent from from MixPrecisionPlugin as well?

Overall idea is similar to ParallelStrategy vs Strategy. If it's parallel specific property, we only included them in ParallelStrategy instead of the base class

File: pytorch_lightning/plugins/precision/mixed.py (hunk 1 above)
Comment (daniellepintz) [reply pytorch_lightning/plugins/precision/mixed.py:25]: this is deprecated in https://github.com/PyTorchLightning/pytorch-lightning/pull/12069

File: pytorch_lightning/plugins/precision/precision_plugin.py (hunk 3 above)
Comment (daniellepintz) [reply pytorch_lightning/plugins/precision/precision_plugin.py:287]: That makes sense! Should I first send out a PR to make DeepSpeedPrecisionPlugin inherit from MixedPrecisionPlugin?

File: pytorch_lightning/plugins/precision/apex_amp.py (hunk 2 above)
Comment (ananthsub) [reply pytorch_lightning/plugins/precision/apex_amp.py:103]: we can't remove it since one is `backend` and the other is `amp_backend` ?

**Review by ananthsub — COMMENTED** [review]
are you planning to fix this as well? https://github.com/PyTorchLightning/pytorch-lightning/blob/d31126c3310af62d76ffb87368ae1e4cc5bbac93/pytorch_lightning/trainer/trainer.py#L2143-L2145
 
this was listed in the linked issue

File: pytorch_lightning/plugins/precision/apex_amp.py (hunk 2 above)
Comment (four4fish) [reply pytorch_lightning/plugins/precision/apex_amp.py:103]: I didn't see anywhere using it and they two are the same thing. @daniellepintz mentioned "this is deprecated in https://github.com/PyTorchLightning/pytorch-lightning/pull/12069"

**Comment by daniellepintz** [comment]
@ananthsub it doesn't seem like `use_amp` is used in the codebase. Can we deprecate it?

File: CHANGELOG.md [hunk 4]
```diff
@@ -146,6 +146,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Added `LayerSync` and `NativeSyncBatchNorm` plugins ([#11754](https://github.com/PyTorchLightning/pytorch-lightning/pull/11754))
 
 
+- Add `precision`, `amp_backend`, and `amp_level` as properties of `PrecisionPlugin` ([#12265](https://github.com/PyTorchLightning/pytorch-lightning/pull/12265))
```
Comment (awaelchli) [inline CHANGELOG.md:149]:
```suggestion
- Added `precision`, `amp_backend`, and `amp_level` as properties of `PrecisionPlugin` ([#12265](https://github.com/PyTorchLightning/pytorch-lightning/pull/12265))
```

File: pytorch_lightning/plugins/precision/apex_amp.py [hunk 5]
```diff
@@ -98,3 +98,11 @@ def state_dict(self) -> Dict[str, Any]:
 
     def load_state_dict(self, state_dict: Dict[str, Any]) -> None:
         amp.load_state_dict(state_dict)
+
+    @property
```
Comment (awaelchli) [inline pytorch_lightning/plugins/precision/apex_amp.py:102]: Can we add the properties to the top please (after init)?

**Comment by four4fish** [comment]
> @ananthsub it doesn't seem like `use_amp` is used in the codebase. Can we deprecate it?

I don't think `Trainer.use_amp` is used anywhere, and "return self.precision == 16" for `Trainer.use_amp` is wrong? I think we have trainer.amp_backend is enough, and `use_amp` can be deprecated.

**Comment by daniellepintz** [comment]
@four4fish sounds good I will send out a PR to deprecate `use_amp`! Also, mind responding to my above question in https://github.com/PyTorchLightning/pytorch-lightning/pull/12265#discussion_r822979865?

File: pytorch_lightning/plugins/precision/precision_plugin.py (hunk 3 above)
Comment (daniellepintz) [reply pytorch_lightning/plugins/precision/precision_plugin.py:287]: Actually one issue with this is we have the `amp_backend` property on the Trainer which references `precision_plugin.amp_backend`. I guess we could have it just first check if the precision plugin is of type MixedPrecisionPlugin and if not return None. what do you think is the best way to go about this?
